tests/ui/pages/desktop/base.py

```diff
@@ -26,6 +26,15 @@ def login(self, email, password):
     def logout(self):
         self.header.click_logout()
 
+    def wait_for_page_to_load(self):
+        body = self.find_element(By.TAG_NAME, 'body')
+        self.wait.until(lambda s: 'firefox' in body.get_attribute('class'))
+
+    @property
+    def wait_for_page(self):
+        url = self.selenium.current_url
+        self.wait.until(lambda _: self.selenium.current_url != url)
```

Comment:
So I am running into the stale element exceotion. I don't see a way to get a new element on the page without waiting until it loads. The only thing I can see to use is the url. I'm all ears to a hint.